packages/ember-toucan-form/src/-private/textarea-field.ts

```diff
@@ -1,33 +1,57 @@
 import Component from '@glimmer/component';
+import { assert } from '@ember/debug';
 
+import type { HeadlessFormBlock, UserData } from './types';
 import type { ToucanFormTextareaFieldComponentSignature as BaseTextareaFieldSignature } from '@crowdstrike/ember-toucan-core/components/form/fields/textarea';
+import type { FormData, FormKey, ValidationError } from 'ember-headless-form';
 
-type ComponentArguments = BaseTextareaFieldSignature['Args'] & {
-  name: string;
+type ComponentArguments<
+  DATA extends UserData,
+  KEY extends FormKey<FormData<DATA>> = FormKey<FormData<DATA>>
+> = Omit<BaseTextareaFieldSignature['Args'], 'error' | 'value' | 'onChange'> & {
```

Comment:
We are setting `'error' | 'value' | 'onChange'` ourselves, so we shouldn't let the user supply them. Only what we are actually passing through should be valid. Instead of `Omit<>` we could also use `Pick<>` to whitelist the passed through arguments, maybe that's better?